### PR TITLE
implemented scanning for subresources

### PR DESCRIPTION
**Pull request by balta3**
Implemented support for scanning and analyzing subresource locators in JAX-RS.

### CODE REVIEW

**Comment by MikeEdgar** [comment]
@balta3 , thanks for the PR. I left a few comments in the code. Please also add several unit tests to help demonstrate the cases this will support and help ensure there are no obvious issues like null pointers, etc.

**Comment by MikeEdgar** [comment]
Fixes #189

File: implementation/src/main/java/io/smallrye/openapi/runtime/scanner/ParameterProcessor.java [hunk 1]
```diff
@@ -1230,6 +1232,22 @@ static boolean isBeanPropertyParam(AnnotationInstance annotation) {
         return relevant;
     }
 
+    /**
+     * Determines if the give method is a JAX-RS sub-resource locator method
```
Comment (EricWittmann) [inline implementation/src/main/java/io/smallrye/openapi/runtime/scanner/ParameterProcessor.java]: "given"? :)

Comment (MikeEdgar) [reply implementation/src/main/java/io/smallrye/openapi/runtime/scanner/ParameterProcessor.java]: Next time... ;-)
